### ISSUE TITLE
How to ignore certain files from the "version bump auto-release" process

### ISSUE THREAD

**Issue by marcelstoer** [danielflower/multi-module-maven-release-plugin#77]
This kind of builds on top of #76. However, since it's likely a missing feature I report it separately.

Scenario
- multi-module project with 20+ modules
- in the root folder I keep a few "management" files alongside the root `pom.xml` like `.editorconfig`, `.gitignore` and `.gitlab-ci.yml`.

Problem: a single change to any of those files will trigger the release of _each module_ upon next invocation of the plugin  as the root project changed.

I argue that this is undesirable in most cases. (How) can this behavior be turned off?

**Comment by marcelstoer** [comment]
@danielflower I would be happy to resource the development of this feature myself but I'll likely need some guidance from you.

**Comment by danielflower** [comment]
Yeah, this is a reasonable feature to have. How about something like a config option called `<ignoredFiles>` or something?

    <ignoredFiles>
        <ignoredFile>README.md</ignoredFile>
        <ignoredFile>docs/whatever.txt</ignoredFile>
    </ignoredFiles>

Could be expanded to be regex or in `.gitignore` format later.

**Comment by marcelstoer** [comment]
Yes, could you also give me a few hints where to integrate it in the code. Which pieces need changing?

**Comment by Marx2** [comment]
this is also a problem with GIT submodules used, as GIT parent project stores hash of commit of each subproject. Commit in subproject have to be followed by commit in parent project.

**Comment by marcelstoer** [comment]
@danielflower I started looking into this and the current hypothesis is that I can work with path filters inside `TreeWalkingDiffDetector`. If you think this is a daft idea then please tell me quickly.

**Comment by talend-jphautin** [comment]
Hello,
I would appreciate this feature to be added as in most project We no have a Jenkinsfile in the root of our repository that is not related to the project itself but to the CI of the project. This is also true when you generate a changelog on the parent project.

**Comment by marcelstoer** [comment]
@talend-jphautin  #94 does address this.

**Comment by Romain125** [comment]
That is a feature that I've been waiting for a few years now. 
I've seen the PR #94 but as it is built on top of #90 @danielflower did not accept it yet.
So...where do we go from here? 
@marcelstoer would it be possible to create a new PR on a fresh master?
Or @danielflower, do you plan ti integrate the #94 ?

Thanks to both of you for your work btw !

**Comment by marcelstoer** [comment]
@Romain125 Sorry, we gave up waiting and moved on.

**Comment by kobynet** [comment]
It shouldn't be too complicated to take @marcelstoer changes into a new branch from master if of-course @marcelstoer agrees to take parts of the code as-is.

**Comment by marcelstoer** [comment]
@kobynet Feel free to do whatever you want with my code.

**Comment by Cavva79** [comment]
Hi @danielflower,
if someone recreate pull request #94 from master, would you merge it?

**Comment by nut-3** [comment]
@danielflower I've recreated the original PR with slight modifications - #152.
I've also added the `requiredPaths` setting, which I think will complement the `ignorePaths` setting.
